components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/certificatemgt/CertificateManagerImpl.java

```diff
@@ -159,33 +162,103 @@ public boolean isConfigured() {
     }
 
     @Override
-    public CertificateMetadataDTO getCertificate(String endpoint, int tenantId) {
-        CertificateMetadataDTO certificateMetadata = null;
+    public List<CertificateMetadataDTO> getCertificates(String endpoint, int tenantId) {
+
+        List<CertificateMetadataDTO> certificateMetadataList = null;
         try {
-            certificateMetadata = certificateMgtDAO.getCertificate("", endpoint, tenantId);
+            certificateMetadataList = certificateMgtDAO.getCertificates("", endpoint, tenantId);
         } catch (CertificateManagementException e) {
             log.error("Error when retrieving certificate metadata for endpoint '" + endpoint + "'", e);
         }
-        return certificateMetadata;
+        return certificateMetadataList;
     }
 
     @Override
     public List<CertificateMetadataDTO> getCertificates(int tenantId) {
+
         List<CertificateMetadataDTO> certificates = null;
         try {
-            certificates = certificateMgtDAO.getCertificates(tenantId);
+            certificates = certificateMgtDAO.getCertificates(null, null, tenantId);
         } catch (CertificateManagementException e) {
             log.error("Error retrieving certificates for the tenantId '" + tenantId + "' ", e);
         }
         return certificates;
     }
 
+    @Override
+    public List<CertificateMetadataDTO> getCertificates(int tenantId, String alias, String endpoint)
+            throws APIManagementException {
+
+        List<CertificateMetadataDTO> certificateMetadataList;
+        try {
+            certificateMetadataList = certificateMgtDAO.getCertificates(alias, endpoint, tenantId);
+        } catch (CertificateManagementException e) {
+            throw new APIManagementException("Error retrieving certificate information for tenantId '" + tenantId +
+                    "' and alias '" + alias + "'");
+        }
+        return certificateMetadataList;
+    }
+
+    @Override
+    public boolean isCertificatePresent(int tenantId, String alias) throws APIManagementException {
+
+        List<CertificateMetadataDTO> certificateMetadataList;
+        try {
+            certificateMetadataList = certificateMgtDAO.getCertificates(alias, null, tenantId);
+        } catch (CertificateManagementException e) {
+            throw new APIManagementException("Error retrieving certificate information for tenantId '" + tenantId +
+                    "' and alias '" + alias + "'");
+        }
+        return certificateMetadataList.size() == 1; // The list would not be null so we check the size.
+    }
+
+    @Override
+    public CertificateInformationDTO getCertificateInformation(String alias) throws APIManagementException {
+
+        try {
+            return certificateMgtUtils.getCertificateInformation(alias);
+        } catch (CertificateManagementException e) {
+            throw new APIManagementException(e);
```

Comment:
Shall we add a meaningful error message before wrapping and throwing to the higher level?, please correct in other places